includes/core/ValueObjects/API/Request/class-create-charge.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * Create charge request.
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Core\ValueObjects\API\Request;
+
+use WCPay\Core\Enums\Http_Methods;
+use WCPay\Core\Enums\Wc_Pay_Endpoints;
+
+/**
+ * Create charge request.
+ */
+final class Create_Charge extends Base_Request {
+
+	/**
+	 * AMount to charge.
+	 *
+	 * @var int $amount
+	 */
+	protected $amount;
+
+	/**
+	 * ID of the source to associate with charge.
+	 *
+	 * @var string $source_id
+	 */
+	protected $source_id;
+
+	/**
+	 * Get amount.
+	 *
+	 * @return int
+	 */
+	public function get_amount() {
+		return $this->amount;
+	}
+
+	/**
+	 * Get source id
+	 *
+	 * @return string
+	 */
+	public function get_source_id() {
+		return $this->source_id;
+	}
```

Comment:
Do we ever need getters for a Request class? 

We'll set the amount and source id and then send the request. In what case would we want to get something from the Request?

Reply:
It cost me like 10 seconds with my IDE to generate getters, and if you need them, you don't have to rely on magic methods and doc block to have a type hinting. They won't do any harm, and sometimes you might need them. The main reason is that I spend like seconds to add them, and it is a bit more clear when you see inside class that you have getters.

Reply:
My point is that the presence of such getter indicates that there's a need for it. I'd say there's no need for getter(or any method) if we don't need it. If we'd be discussion parent class that provides functionality that be a different case but here I think the presence of these methods is misleading.